File: Convert Binary Number in a Linked List to Integer/ConvertBinaryNumberinaLinkedListtoInteger.java
public class ConvertBinaryNumberinaLinkedListtoInteger {
    public static void main(String[] args) {
        ListNode h = new ListNode(1, new ListNode(0, new ListNode(1)));
        getDecimalValue(h);

    }

    public static class ListNode {
        int val;
        ListNode next;

        ListNode() {
        }

        ListNode(int val) {
            this.val = val;
        }

        ListNode(int val, ListNode next) {
            this.val = val;
            this.next = next;

        }
    }

    public static int getDecimalValue(ListNode head) {
        int binNumber = 0;
        while(head!=null){
            binNumber = binNumber*2 + head.val;
            head = head.next;
        }
        System.out.println(binNumber);

        return 1;

    }
}

File: Convert Binary Number in a Linked List to Integer/ConvertBinaryNumberinaLinkedListtoInteger.cpp
#include<iostream>
using namespace std;

struct ListNode
{
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr){}
    ListNode(int x) : val(x), next(nullptr){}
    ListNode(int x, ListNode *next) : val(x), next(next){}
    
};

int getDecimalValue(ListNode* head) {
    int ans = 0;
    while(head!=nullptr){
        ans = ans*2 + head->val;
        head = head->next;
    }
    return ans;
}


int main(){
    ListNode* head = new ListNode(1, new ListNode(0, new ListNode(1)));
    int ans = getDecimalValue(head);
    cout<<ans;
    cout<<"\n";
    return 0;
}
